test/gtest/uct/test_md.cc

```diff
@@ -1150,31 +1153,19 @@ class test_md_non_blocking : public test_md
 UCS_TEST_SKIP_COND_P(test_md_non_blocking, reg_advise,
                      !check_caps(UCT_MD_FLAG_REG | UCT_MD_FLAG_ADVISE))
 {
-    test_reg_advise(UCS_KBYTE, UCS_KBYTE, 0, true);
-    test_reg_advise(UCS_KBYTE, UCS_KBYTE / 2, 0, true);
-    test_reg_advise(UCS_KBYTE, UCS_KBYTE / 2, UCS_KBYTE / 4, true);
-
-    /*
-     * TODO: These tests should be enabled
-     * when https://redmine.mellanox.com/issues/336376 fixed
-
-     * test_reg_advise(UCS_MBYTE, UCS_MBYTE, 0, true);
-     * test_reg_advise(UCS_MBYTE, UCS_MBYTE / 2, 0, true);
-     * test_reg_advise(UCS_MBYTE, UCS_MBYTE / 2, UCS_MBYTE / 4, true);
-     */
+    for (auto size: {UCS_KBYTE, UCS_MBYTE}) {
+        test_reg_advise(size, size, 0, true);
+        test_reg_advise(size, size / 2, 0, true);
+        test_reg_advise(size, size / 2, size / 4, true);
+    }
 }
 
 UCS_TEST_SKIP_COND_P(test_md_non_blocking, reg,
                      !check_caps(UCT_MD_FLAG_REG))
 {
-    test_reg_advise(UCS_KBYTE, 0, 0, true);
-
-    /*
-     * TODO: This test should be enabled
-     * when https://redmine.mellanox.com/issues/336376 fixed
-
-     * test_reg_advise(UCS_MBYTE, 0, 0, true);
-     */
+    for (auto size: {UCS_KBYTE, UCS_MBYTE}) {
+        test_reg_advise(size, 0, 0, true);
+    }
```

Comment:
this can be a member of `test_md` and `test_ib_md` could also use it.

Reply:
I am not sure that I got your point. You propose to create new function in `test_md` that run `test_reg_advise` for different sizes and call it from `test_ib_md`?

Reply:
you have wholly copy pasted tests in two classes. Imo, making the test a member function in a base class is worth it

Reply:
Unified